scale/ingest/models.py

```diff
@@ -723,7 +721,44 @@ def get_details(self, scan_id):
         :rtype: :class:`ingest.models.Scan`
         """
 
-        return Scan.objects.select_related('job', 'job__job_type').get(pk=scan_id)
+        scan = Scan.objects.select_related('job', 'job__job_type')
+        scan = scan.select_related('dry_run_job', 'dry_run_job__job_type')
+        scan = scan.get(pk=scan_id)
+
+        return scan
+
+    @transaction.atomic
+    def queue_scan(self, scan_id, dry_run=True):
+        """Retrieves a Scan model and uses metadata to place a job to run the
+        Scan process on the queue. All changes to the database will occur in an
+        atomic transaction.
+
+        :param scan_id: The unique identifier of the Scan process.
+        :type scan_id: int
+        :param dry_run: Whether the scan will execute as a dry run
+        :type dry_run: bool
+        :returns: The new Scan process
+        :rtype: :class:`ingest.models.Scan`
+        """
+
```

Comment:
Dry runs are allowed to be overwritten, normal scans that perform ingests cannot be re-run.